web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,360 @@
+      type: pageData.contributionType,
+      contributorsStats: [],
+      dateFrom: null,
+      dateUntil: null,
+      startDate: null,
+      endDate: null,
+    };
+  },
+  mounted() {
+    this.fetchGraphData();
+
+    $('.ui.dropdown').dropdown({
+      onChange: (val) => {
+        this.dateFrom = this.startDate;
+        this.dateUntil = this.endDate;
+        this.type = val;
+        this.sortContributors();
+      }
+    });
+  },
+  methods: {
+    sortContributors() {
+      const contributors = this.filterContributorWeeksByDateRange();
+      const sortingCriteria = `total_${this.type}`;
+      this.sortedContributors = Object.values(contributors).filter((contributor) => contributor[sortingCriteria] !== 0).sort((a, b) =>
+        a[sortingCriteria] > b[sortingCriteria] ? -1 : a[sortingCriteria] === b[sortingCriteria] ? 0 : 1
+      ).slice(0, 100);
+    },
+    async fetchGraphData() {
+      this.isLoading = true;
+      try {
+        const response = await fetch(`${this.repoLink}/activity/contributors/data`);
+        const data = await response.json();
+        const {total, ...rest} = data;
+        this.contributorsStats = rest;
+        this.dateFrom = new Date(total.weeks[0].week);
+        this.dateUntil = new Date();
+        this.startDate = this.dateFrom;
+        this.endDate = this.dateUntil;
+        this.sortContributors();
+        this.totalStats = total;
+        this.errorText = '';
+      } catch (e) {
+        this.errorText = e.message;
+      } finally {
+        this.isLoading = false;
+      }
+    },
+
+    filterContributorWeeksByDateRange() {
+      const filteredData = {};
+
+      const data = this.contributorsStats;
+      for (const key of Object.keys(data)) {
+        const user = data[key];
+        user['total_commits'] = 0;
+        user['total_additions'] = 0;
+        user['total_deletions'] = 0;
+        user['max_contribution_type'] = 0;
+        const filteredWeeks = user.weeks.filter((week) => {
+          const weekDate = new Date(week.week);
+          if (weekDate >= this.dateFrom && weekDate <= this.dateUntil) {
+            user['total_commits'] += week.commits;
+            user['total_additions'] += week.additions;
+            user['total_deletions'] += week.deletions;
+            if (week[this.type] > user['max_contribution_type']) {
+              user['max_contribution_type'] = week[this.type];
+            }
+            return true;
+          }
+          return false;
+        });
+
+        filteredData[key] = {...user, weeks: filteredWeeks};
+      }
+
+      return filteredData;
+    },
+
+    maxMainGraph() {
+      const maxValue = Math.max(
+        ...this.totalStats.weeks.map((o) => o[this.type])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
```

Comment:
lol, yeah you are right :D This method is there to calculate the maximum Y value for the graphs. Chartjs can automatically calculate it, and it can calculate it better than me but the problem is if let Chartjs do it, it will also change max Y value when I zoom on the graphs. I want them to be fixed, I couldn't find a way to make them fixed, hence the method...

Reply:
Added comments to code.